localstack-core/localstack/services/cloudwatch/provider_v2.py

```diff
@@ -835,6 +831,84 @@ def _get_timestamp(input: dict):
             history = [h for h in history if (date := _get_timestamp(h)) and date <= end_date]
         return DescribeAlarmHistoryOutput(AlarmHistoryItems=history)
 
+    def _evaluate_composite_alarms(self, context: RequestContext, triggering_alarm):
+        store = self.get_store(context.account_id, context.region)
+        alarms = list(store.alarms.values())
+        composite_alarms = [a for a in alarms if isinstance(a, LocalStackCompositeAlarm)]
+        for composite_alarm in composite_alarms:
+            new_state_value = StateValue.OK
+            alarm_rule = composite_alarm.alarm["AlarmRule"]
+            # assuming that a rule consists only of ALARM evaluations of metric alarms, with OR logic applied
+            metric_alarm_arns = re.findall(
```

Comment:
Alarm rule parsing and further evaluation definitely deserves more checks, like check if parsed metric alarm ARM exists in the store and actually check for `OR` presence. However, I wanted to first get overall feedback on PR and approach and then work on these details.

Reply:
So there are some things that can break in real usage if unchecked, but I'd like to fix them after a general feedback on approach.

Reply:
Added several input validations in `put_composite_alarm`, namely:
- if each metric alarm condition starts with `ALARM`
- if there is a metric alarm in the store that corresponds to a referenced one

Likely some edge cases are still left out. For a more robust and extensible parsing likely a different approach needs to be taken, probably with recursive descent parsing.